File: src/main/java/domain/math/RescaleAxis.java
package domain.math;

public class RescaleAxis {
    public static Double[] rescaleAxis(Double[] axis, Double max) {

        double[] scalar = new double[axis.length];

        double range = Math.abs(max - axis[0]);

        for(int i = 0; i < axis.length; i++) {
            scalar[i] = (axis[i] - axis[0])/ (axis[axis.length - 1] - axis[0]);
        }

        Double[] newAxis = new Double[axis.length];

        for(int i = 0; i < axis.length; i++) {
            if(i == 0) {
                newAxis[0] = axis[0];
            } else {
                newAxis[i] = axis[0] + (scalar[i] * range);
            }
        }

        return newAxis;
    }
}

File: src/main/kotlin/domain/math/RescaleAxis.kt
package domain.math

import kotlin.math.abs

object RescaleAxis {
    fun rescaleAxis(axis: Array<Double>, max: Double): Array<Double> {
        val scalar = DoubleArray(axis.size)
        val range = abs(max - axis[0])

        for (i in axis.indices) {
            scalar[i] = (axis[i] - axis[0]) / (axis[axis.size - 1] - axis[0])
        }

        return Array(axis.size) { i ->
            if (i == 0) axis[0]
            else axis[0] + scalar[i] * range
        }
    }
}
